### PR TITLE
add MultiClusterConfig and auto manifest scripts

### PR DESCRIPTION
**Pull request by luolanzone**
1. add new CRD MultiClusterConfig as custom config setting, refer to [kubebuilder component-config](https://book.kubebuilder.io/component-config-tutorial/define-config.html)
2. add manifests auto generating scripts
3. update Makefile and build folder
Signed-off-by: Lan Luo <luola@vmware.com>

### CODE REVIEW

File: multicluster/apis/multicluster/v1alpha1/multiclusterconfig_types.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	config "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Comment (abhiraut) [inline multicluster/apis/multicluster/v1alpha1/multiclusterconfig_types.go]: maybe remove this

File: multicluster/cmd/multicluster-controller/controller.go [hunk 2]
```diff
@@ -48,122 +45,102 @@ func init() {
...
-	flag.StringVar(&config, "config", "", "Configuration file path.")
-
+func run(o *Options) error {
 	opts := zap.Options{
 		Development: true,
 	}
-	opts.BindFlags(flag.CommandLine)
-	flag.Parse()
-
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
```
Comment (abhiraut) [inline multicluster/cmd/multicluster-controller/controller.go:52]: not part of your PR; but can we not use klog?

File: multicluster/cmd/multicluster-controller/options.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+func (o *Options) setDefaults() {
+	o.options = ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     ":8080",
+		Port:                   9443,
+		HealthProbeBindAddress: ":8081",
+		LeaderElection:         false,
+		LeaderElectionID:       "6536456a.crd.antrea.io",
```
Comment (abhiraut) [inline multicluster/cmd/multicluster-controller/options.go]: how is this ID generated? if default for leader election is false, then do we need election ID?

File: multicluster/hack/generate-manifest.sh [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+fi
+
+if [ "$LEADER" ==  false ] && [ "$MEMBER" == true ]; then
+    sed -i.bak -E "s/member: false/member: true/" controller_manager_config.yaml
+fi
+
+cp $KUSTOMIZATION_DIR/default/configmap/kustomization.yaml kustomization.yaml
+$KUSTOMIZE build
+
+rm -rf $TMP_DIR
```
Comment (abhiraut) [inline multicluster/hack/generate-manifest.sh]: add newline

File: multicluster/cmd/multicluster-controller/controller.go (hunk 2 above)
Comment (luolanzone) [reply multicluster/cmd/multicluster-controller/controller.go:52]: yes, I think we can.

File: multicluster/cmd/multicluster-controller/options.go (hunk 3 above)
Comment (luolanzone) [reply multicluster/cmd/multicluster-controller/options.go]: it's automatically generated by kubebuilder,I just moved it from previous main.go, I think when `LeaderElection` is false, this field should be useless. I can remove this line.

File: multicluster/hack/generate-manifest.sh (hunk 4 above)
Comment (luolanzone) [reply multicluster/hack/generate-manifest.sh]: done

File: multicluster/cmd/multicluster-controller/options.go (hunk 3 above)
Comment (abhiraut) [reply multicluster/cmd/multicluster-controller/options.go]: sounds good

**Pull request merged**